src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonedState = { ...state };
+
+  const allState = [];
+
+  for (const action of actions) {
+    transform(clonedState, action);
+
+    const clone = { ...clonedState };
+
+    allState.push(clone);
```

Comment:
clone is redundant variable 
```suggestion
    transform(clonedState, action);
    allState.push({ ...clonedState });
```